### PR TITLE
Support TLS connections for Postgres sources

### PR DESCRIPTION
**Pull request by uce**
Opening this as a draft while I work on the `rust-postgres` client changes that block the remaining changes. @petrosagg Do the tests and the places where we create the TLS connector look sane?

## TODO
- [x] Use existing test certs infra (just saw this after opening the PR)
- [ ] Add remaining sslmodes to `rust-postgres`

### CODE REVIEW

**Comment by petrosagg** [comment]
Yes, this looks great! One thing that comes to mind is that we don't need to duplicate the full testdrive tests for both the ssl and non-ssl version. We just need two small tests testing the connection modes and then run the full suite that verify the correctness of the data with SSL.

Also, what do you think of finishing and merging the support for the three modes already supported (Disable, Prefer, Require) before implementing the more complex ones? This will allow users actively testing this feature to use it with RDS databases

**Comment by uce** [comment]
I updated the PR.

> Yes, this looks great! One thing that comes to mind is that we don't need to duplicate the full testdrive tests for both the ssl and non-ssl version. We just need two small tests testing the connection modes and then run the full suite that verify the correctness of the data with SSL.

As discussed offline, we kept the tests as they are and moved the existing tests to require SSL.

> Also, what do you think of finishing and merging the support for the three modes already supported (Disable, Prefer, Require) before implementing the more complex ones? 

I updated the PR. Unfortunately, I didn't get the test-certs to work directly (because of file perms), but didn't want to block the PR on it. I might give it another shot after this is merged.

File: src/postgres-util/src/lib.rs
Comment (petrosagg) [inline src/postgres-util/src/lib.rs:80]: Is this type annotation necessary?

File: test/pg-cdc/mzcompose.yml
Comment (petrosagg) [inline test/pg-cdc/mzcompose.yml:72]: I'm not sure if it would make this setup simpler, but the postgres docker image supports adding initialization scripts in it that run automatically. `rust-postgres` uses this mechanism to set up certificates, `pg_hba.conf`, and users.

This is the script https://github.com/sfackler/rust-postgres/blob/master/docker/sql_setup.sh which is then imported here https://github.com/sfackler/rust-postgres/blob/master/docker/Dockerfile#L3

**Review by petrosagg — APPROVED** [review]
This looks great! thank you

File: src/postgres-util/src/lib.rs
Comment (uce) [reply src/postgres-util/src/lib.rs:80]: That was a left-over. Thanks for catching 👍🏼

File: test/pg-cdc/mzcompose.yml
Comment (uce) [reply test/pg-cdc/mzcompose.yml:72]: Nice! I hadn't seen this before. I think it will make things quite a bit simpler in combination with mzbuild (assuming I can depend on `test-certs` directly). Let me try that real quick and update if possible.

Comment (philip-stoev) [reply test/pg-cdc/mzcompose.yml:72]: You can also check out ```test/kafka-ssl/mzcompose.yml``` where the testdrive container entrypoint is used to put certs into position.

**Review by philip-stoev — APPROVED** [review]
Thank you for the tests!

File: test/pg-cdc/mzcompose.yml
Comment (uce) [reply test/pg-cdc/mzcompose.yml]: As discussed offline, the approach from `test/kafka-ssl/mzcompose.yml` does not carry over, because we run testdrive after postgres has started.

It took a bit longer than expected, but I was able to simplify the `mzcompose.yml` by introducing a `postgres` mzimage.

**Pull request merged**
